### PR TITLE
fix: Fix TopoJSON encoding issue

### PR DESCRIPTION
**Pull request by emanuelaepure10**
Use Jackson's ObjectMapper to parse the JSON string into a JsonNode, then recursively traverses the JsonNode to replace "null" values with null. Finally, it converts the modified JsonNode back to a JSON string.

ING-3979

### CODE REVIEW

File: io/plugins/eu.esdihumboldt.hale.io.topojson/src/eu/esdihumboldt/hale/io/topojson/TopoJsonInstanceWriter.java [hunk 1]
```diff
@@ -171,8 +177,60 @@ private void convertShapefileToTopoJson(String shapefilePath,
...
+
+			if (compressOutput) {
+				Toolbox.writeFile(targetPath, Compress.compressB64(modifiedJsonString));
+			}
+			else {
+				Toolbox.writeFile(targetPath, modifiedJsonString);
+			}
+
+		} catch (Exception e) {
+			e.printStackTrace();
```
Comment (stempler) [inline io/plugins/eu.esdihumboldt.hale.io.topojson/src/eu/esdihumboldt/hale/io/topojson/TopoJsonInstanceWriter.java:201]: The exception should not be dropped silently but end up in the report and ideally make it fail. You should be able to achieve that by letting this method throw it.

File: io/plugins/eu.esdihumboldt.hale.io.topojson/src/eu/esdihumboldt/hale/io/topojson/TopoJsonInstanceWriter.java [hunk 2]
```diff
@@ -171,8 +177,60 @@ private void convertShapefileToTopoJson(String shapefilePath,
...
+	/**
+	 * @param node to replace the unwanted characters
+	 * @return the clean node
+	 */
+	public static JsonNode replaceUnwantedCharacters(JsonNode node) {
+		if (node.isObject()) {
+			ObjectNode objectNode = (ObjectNode) node;
+			objectNode.fields().forEachRemaining(entry -> {
+				JsonNode value = entry.getValue();
+				if (value.isTextual() && value.textValue().contains("\u0000")) {
```
Comment (stempler) [inline io/plugins/eu.esdihumboldt.hale.io.topojson/src/eu/esdihumboldt/hale/io/topojson/TopoJsonInstanceWriter.java:214]: What about the case with the null date?

Comment (emanuelaepure10) [reply io/plugins/eu.esdihumboldt.hale.io.topojson/src/eu/esdihumboldt/hale/io/topojson/TopoJsonInstanceWriter.java:214]: That's missing at the moment. 
I'm uncertain about the best approach to resolve the date issue.

File: io/plugins/eu.esdihumboldt.hale.io.topojson/src/eu/esdihumboldt/hale/io/topojson/TopoJsonInstanceWriter.java [hunk 3]
```diff
@@ -169,10 +181,110 @@ private List<URI> writeIntermediateShapefiles(InstanceCollection instances,
...
+									objectNode.put(entry.getKey(), textValue);
+								}
+
+							}
+							else {
+								objectNode.put(entry.getKey(), textValue);
+							}
+
+						} catch (Exception e) {
+							objectNode.put(entry.getKey(), (String) null);
```
Comment (stempler) [inline io/plugins/eu.esdihumboldt.hale.io.topojson/src/eu/esdihumboldt/hale/io/topojson/TopoJsonInstanceWriter.java:261]: I think it would be good to log this exception, since it seems to be unexpected that it may happen (since date parsing errors are handled somewhere else).

File: io/plugins/eu.esdihumboldt.hale.io.topojson.test/src/eu/esdihumboldt/hale/io/topojson/test/TopoJsonInstanceWriterTest.groovy [hunk 4]
```diff
@@ -56,6 +57,7 @@ class TopoJsonInstanceWriterTest {
 		SimpleType {
 			id(String)
 			name(String)
+			label(String)
```
Comment (stempler) [inline io/plugins/eu.esdihumboldt.hale.io.topojson.test/src/eu/esdihumboldt/hale/io/topojson/test/TopoJsonInstanceWriterTest.groovy:60]: Having an additional unit test to test the case with the null date would be great.

File: io/plugins/eu.esdihumboldt.hale.io.topojson.test/src/eu/esdihumboldt/hale/io/topojson/test/TopoJsonInstanceWriterTest.groovy [hunk 5]
```diff
@@ -144,8 +155,9 @@ class TopoJsonInstanceWriterTest {
 			assertEquals(1, json.objects.Topology.geometries[1].arcs.size())
 			assertEquals(3, json.objects.Topology.geometries[1].arcs[0].size())
 			assertEquals([-1, 2, 3], json.objects.Topology.geometries[1].arcs[0])
-		}
 
+			assertNull(json.objects.Topology.geometries[1].'properties'.label)
+			assertNull(json.objects.Topology.geometries[1].'properties'.date)
```
Comment (stempler) [inline io/plugins/eu.esdihumboldt.hale.io.topojson.test/src/eu/esdihumboldt/hale/io/topojson/test/TopoJsonInstanceWriterTest.groovy:160]: Please also check the case where a value should be present, to test that that value is not replaced by null unintentionally.

**Review by stempler — APPROVED** [review]
